physcraper/__init__.py

```diff
@@ -2429,68 +2422,59 @@ def remove_identical_seqs(self):
+                            # else:
+                            #     debug(self.mrca_ncbi)
+                            #     debug("should not happen. think about something to do!")
+                            #     mrca_ncbi = None
+                            # # debug(stop)
+                            # # debug(mrca_ncbi)
+
+
+                            # rank_mrca_ncbi = self.ids.ncbi_parser.get_rank(mrca_ncbi)
                             # get rank to delimit seq to ingroup_mrca
-                            # get name first
-                            ncbi_id = None
-                            tax_name = None
-                            if gb_id[:6] == "unpubl":
-                                debug("unpubl data")
-                                # debug(self.data.gb_dict[gb_id])
-                                tax_name = self.data.gb_dict[gb_id][u"^ot:ottTaxonName"]
-                                ncbi_id = self.data.gb_dict[gb_id][u"^ncbi:taxon"]
-                                if tax_name is None:
-                                    tax_name = self.data.gb_dict[gb_id][u'^user:TaxonName']
-                                if ncbi_id is None:
-                                    # debug(tax_name.split(" ")[0])
-                                    tax_lin_name = tax_name.split(" ")[0]
-                                    tax_lin_name = tax_lin_name.split("_")[0]
-                                    # debug(tax_lin_name)
-                                    ncbi_id = self.ids.ncbi_parser.get_id_from_name(tax_lin_name)
-                            elif len(gb_id.split(".")) >= 2:
-                                if gb_id in self.data.gb_dict.keys() and 'staxids' in self.data.gb_dict[gb_id].keys():
-                                    tax_name = self.data.gb_dict[gb_id]['sscinames']
-                                    ncbi_id = self.data.gb_dict[gb_id]['staxids']
+                            ncbi_id, tax_name = get_taxinfo_from_acc(gb_id, self.data, self.ids)
+
+
+                            rank_mrca_ncbi = set()
+                            print(self.mrca_ncbi_list)
+                            # print(stop)
+                            for mrca in self.mrca_ncbi_list:
+                                debug("mrca")
+                                debug(mrca)
+                                rank_mrca = self.ids.ncbi_parser.get_rank(mrca)
+                                rank_mrca_ncbi.add(rank_mrca)
+
+                                try:  # sometimes ncbi has wrong id linked
+                                    input_rank_id = self.ids.ncbi_parser.match_id_to_mrca(ncbi_id, mrca)
+                                except:  # this is for the wrong ncbi link, get right tax_id and do search again
+                                    debug("wrong tax_id given by ncbi? tax_id: {}, tax_name:{}".format(ncbi_id, tax_name))
+                                    ncbi_id = self.ids.ncbi_parser.get_id_from_name(tax_name)
```

Comment:
We should make sure that we are updating the stored "^ncbi:taxon" info fo rthat tip to match the one we used.